### PR TITLE
Move toLocaleString testing to intl402

### PR DESCRIPTION
**Pull request by FrankYFTang**
Move part of the test of toLocaleString which depends on
15 Amendments to the ECMAScript® 2021 Internationalization API Specification
to intl402. Keep behavior specified in earlier chapters in built-ins

@Ms2ger @jugglinmike @ryzokuken @ljharb @ptomato @pipobscure

### CODE REVIEW

**Comment by rwaldron** [comment]
This makes sense to me, but I want one other person (from the summon list) to sign-off

**Review by ptomato — DISMISSED** [review]
LGTM, though maybe the ones remaining in built-ins should be renamed to something like `returns-string.js` since they no longer test conflicting options?

**Review by ptomato — DISMISSED** [review]

**Comment by jugglinmike** [comment]
The pull request description makes sense to me, as does @ptomato's suggestion, but I think the patch is erroneous.

I expected the return type assertions to be persisted in the `test/built-ins/` directory (renamed as Philip recommended). That appears to be done.

I also expected the assertions for abrupt completions to be relocated to the `test/intl402/` directory. Instead, it looks like the return type assertions have been copied there. The corresponding files between the two directories are identical:

    $ git diff --diff-filter A main... --name-only | xargs sha1sum | sort
    06bb829d1109f16571280cc8029409f172821d9e  test/built-ins/Temporal/PlainDate/prototype/toLocaleString/return-string.js
    06bb829d1109f16571280cc8029409f172821d9e  test/intl402/Temporal/PlainDate/prototype/toLocaleString/options-conflict.js
    1a2112323878169f18ace50e3b201a745a87cf42  test/built-ins/Temporal/Instant/prototype/toLocaleString/return-string.js
    1a2112323878169f18ace50e3b201a745a87cf42  test/intl402/Temporal/Instant/prototype/toLocaleString/options-conflict.js
    33a813ac7e3c1492d14c40b86b9774f44dfe7787  test/built-ins/Temporal/ZonedDateTime/prototype/toLocaleString/return-string.js
    33a813ac7e3c1492d14c40b86b9774f44dfe7787  test/intl402/Temporal/ZonedDateTime/prototype/toLocaleString/options-conflict.js
    c0403ca0bcb74122726caaac442b7bfec063f592  test/built-ins/Temporal/PlainDateTime/prototype/toLocaleString/return-string.js
    c0403ca0bcb74122726caaac442b7bfec063f592  test/intl402/Temporal/PlainDateTime/prototype/toLocaleString/options-conflict.js
    cf023db6e9386835a806e38d419a235773af9b29  test/built-ins/Temporal/PlainTime/prototype/toLocaleString/return-string.js
    cf023db6e9386835a806e38d419a235773af9b29  test/intl402/Temporal/PlainTime/prototype/toLocaleString/options-conflict.js

Is that what was intended?

**Comment by FrankYFTang** [comment]
> The pull request description makes sense to me, as does @ptomato's suggestion, but I think the patch is erroneous.
> 
> I expected the return type assertions to be persisted in the `test/built-ins/` directory (renamed as Philip recommended). That appears to be done.
> 
> I also expected the assertions for abrupt completions to be relocated to the `test/intl402/` directory. Instead, it looks like the return type assertions have been copied there. The corresponding files between the two directories are identical:
> 
> ```
> $ git diff --diff-filter A main... --name-only | xargs sha1sum | sort
> 06bb829d1109f16571280cc8029409f172821d9e  test/built-ins/Temporal/PlainDate/prototype/toLocaleString/return-string.js
> 06bb829d1109f16571280cc8029409f172821d9e  test/intl402/Temporal/PlainDate/prototype/toLocaleString/options-conflict.js
> 1a2112323878169f18ace50e3b201a745a87cf42  test/built-ins/Temporal/Instant/prototype/toLocaleString/return-string.js
> 1a2112323878169f18ace50e3b201a745a87cf42  test/intl402/Temporal/Instant/prototype/toLocaleString/options-conflict.js
> 33a813ac7e3c1492d14c40b86b9774f44dfe7787  test/built-ins/Temporal/ZonedDateTime/prototype/toLocaleString/return-string.js
> 33a813ac7e3c1492d14c40b86b9774f44dfe7787  test/intl402/Temporal/ZonedDateTime/prototype/toLocaleString/options-conflict.js
> c0403ca0bcb74122726caaac442b7bfec063f592  test/built-ins/Temporal/PlainDateTime/prototype/toLocaleString/return-string.js
> c0403ca0bcb74122726caaac442b7bfec063f592  test/intl402/Temporal/PlainDateTime/prototype/toLocaleString/options-conflict.js
> cf023db6e9386835a806e38d419a235773af9b29  test/built-ins/Temporal/PlainTime/prototype/toLocaleString/return-string.js
> cf023db6e9386835a806e38d419a235773af9b29  test/intl402/Temporal/PlainTime/prototype/toLocaleString/options-conflict.js
> ```
> 
> Is that what was intended?

hum... I think I made some mistake when I intend to move some file and somehow landed the modified copies instead of the origional copies. Need to fix that. Sorry

**Comment by rwaldron** [comment]
@FrankYFTang how's it going here?

**Comment by FrankYFTang** [comment]
sorry, forget about this. fix it. PTAL

File: test/built-ins/Temporal/Instant/prototype/toLocaleString/return-string.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright (C) 2021 Kate Miháliková. All rights reserved.
```
Comment (rwaldron) [inline test/built-ins/Temporal/Instant/prototype/toLocaleString/return-string.js:1]: @FrankYFTang this is not your name ;)

**Review by rwaldron — CHANGES_REQUESTED** [review]

File: test/built-ins/Temporal/Instant/prototype/toLocaleString/return-string.js (hunk 1 above)
Comment (FrankYFTang) [reply test/built-ins/Temporal/Instant/prototype/toLocaleString/return-string.js:1]: It is a strip down version from 
https://github.com/tc39/test262/blob/main/test/built-ins/Temporal/Instant/prototype/toLocaleString/options-conflict.js
so I have keep the credit to the author of that file

**Review by rwaldron — APPROVED** [review]

**Pull request merged**